src/java.base/share/classes/java/io/ObjectInputStream.java

```diff
@@ -357,27 +357,13 @@ private static class Logging {
      */
     private boolean streamFilterSet;
 
-    /**
-     * cache LUDCL (Latest User Defined Class Loader) till completion of
-     * read* requests
-     */
-
-    @SuppressWarnings("removal")
-    private static final boolean isClassCachingEnabled =
-            AccessController.doPrivileged(new GetClassCachingSettingAction());
+    /* cache LUDCL (Latest User Defined Class Loader) till completion of read requests */
+    /* if true LUDCL/forName results would be cached, true by default starting Java8 */
+    private static final boolean isClassCachingEnabled = Boolean.parseBoolean(System.getProperty("com.ibm.enableClassCaching", "true"));
     /* ClassByNameCache Entry for caching class.forName results upon enableClassCaching */
     private static final ClassByNameCache classByNameCache =
             isClassCachingEnabled ? new ClassByNameCache() : null;
 
-    /** if true LUDCL/forName results would be cached, true by default starting Java8 */
-    private static final class GetClassCachingSettingAction
```

Comment:
While you're here, please delete lines 488, 490 and 658; and correct the indentation on lines 550, 552 and 553.
The redundant initializers for `refreshLudcl` and `startingLudclObject` could be removed.

Reply:
Removed the lines, initialization, and extra spaces.